File: backend/Models/TodoItem.cs
namespace backend.Models;

public class TodoItem
{
    public int Id { get; set; }

    public string Title { get; set; } = string.Empty;

    public string? Description { get; set; }

    public bool IsCompleted { get; set; } = false;

    // Foreign Keys
    public int CategoryId { get; set; }

    public int PriorityId { get; set; }

    // Navigation Properties
    public Category Category { get; set; } = null!;

    public Priority Priority { get; set; } = null!;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/src/interfaces/TodoItem.ts
import type { Category } from './Category';
import type { Priority } from './Priority';

export interface TodoItem {
  id: number;
  title: string;
  description: string;
  categoryId: number;
  category?: Category;
  priorityId: number;
  priority?: Priority;
  isCompleted: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateTodoDto {
  title: string;
  description: string;
  categoryId: number;
  priorityId: number;
}

export interface UpdateTodoDto {
  title: string;
  description: string;
  categoryId: number;
  priorityId: number;
  isCompleted: boolean;
}
